examples/gadgets/trace/exec/exec.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+
+	"github.com/cilium/ebpf/rlimit"
+
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/exec/tracer"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/exec/types"
+)
+
+func main() {
+	// In some kernel versions it's needed to bump the rlimits to
+	// use run BPF programs.
+	if err := rlimit.RemoveMemlock(); err != nil {
+		return
+	}
+
+	// Define a callback to be called each time there is an event.
+	eventCallback := func(event types.Event) {
+		fmt.Printf("A new %q process with pid %d was executed\n",
+			event.Comm, event.Pid)
+	}
+
+	// Create the tracer. An empty configuration is passed as we are
+	// not interesting on filtering by any container. For the same
+	// reason, no enricher is passed.
+	tracer, err := tracer.NewTracer(&tracer.Config{}, nil, eventCallback)
```

Comment:
BTW, it would be nice if we could pass `nil` instead of `&tracer.Config{}`.

Reply:
Good point. I was thinking that we could make the `enricher` and `callback` part of the configuration, then it will only receive a single parameter making it easier to extend.

Reply:
Yeah, that could also resolve [this discussion](https://github.com/kinvolk/inspektor-gadget/pull/775/files#r931213502).